js/interface.js

```diff
@@ -138,9 +182,11 @@ dsQueryProvider.then(function onForwardDsQueryProvider(result) {
   };
   Fliplet.DataSources.update(options)
     .then(function() {
-      data = {
-        type: result.data.selectedModeIdx === 1 ? 'sms' : 'email',
-        dataSourceQuery: result.data
+      data.dataSourceQuery = result.data;
+
+      // Email verification type is choosed on ds query provider
+      if (data.type !== 'sms') {
+        data.type = result.data.selectedModeIdx === 1 ? 'domain' : 'email';
```

Comment:
When the build calls the send-validation endpoint it needs to know if it's regular validation or with email domain white listing.